pipelines/wan2_1/blocks/text_conditioning.py

```diff
@@ -48,55 +60,120 @@ def intermediate_outputs(self) -> list[OutputParam]:
+                prompts_to_encode = target_prompts
 
-        # Only run text_encoder if prompt changed
-        if (
+        # Normalize prompts to list[dict] format for processing
+        normalized_prompts = self._normalize_prompts(prompts_to_encode)
+
+        # Check if prompts changed (compare normalized versions)
+        # Use prompts_to_encode (which may be target_prompts) for comparison
+        prompts_changed = (
             block_state.current_prompts is None
-            or block_state.current_prompts != block_state.prompts
+            or self._normalize_prompts(block_state.current_prompts)
+            != normalized_prompts
+        )
+
+        with torch.autocast(
+            str(components.config.device), dtype=components.config.dtype
         ):
-            with torch.autocast(
-                str(components.config.device), dtype=components.config.dtype
-            ):
-                conditional_dict = components.text_encoder(
-                    text_prompts=[block_state.prompts]
-                    if isinstance(block_state.prompts, str)
-                    else block_state.prompts
-                )
-            block_state.prompt_embeds = conditional_dict["prompt_embeds"]
-            block_state.current_prompts = block_state.prompts
-            block_state.prompt_embeds_updated = True
+
+            def encode_prompt_items(
+                prompt_items: list[dict],
+            ) -> tuple[list[torch.Tensor], list[float]]:
+                """Encode a list of prompt dicts into embeddings and weights."""
+                if not prompt_items:
+                    return [], []
+
+                # Extract texts and weights
+                texts = [item.get("text", "") for item in prompt_items]
+                weights = [item.get("weight", 1.0) for item in prompt_items]
+
+                # Batch encode all prompts at once
+                conditional_dict = components.text_encoder(text_prompts=texts)
+                batched_embeds = conditional_dict["prompt_embeds"]
+
+                # Split batched tensor into individual tensors, preserving batch dimension
+                # batched_embeds shape: [batch_size, seq_len, hidden_dim]
+                # Each embedding should be [1, seq_len, hidden_dim] to match original behavior
+                embeddings = [
+                    batched_embeds[i].unsqueeze(0)
+                    for i in range(batched_embeds.shape[0])
+                ]
+
+                return embeddings, weights
+
+            # Encode regular prompts if they changed
+            if prompts_changed:
+                (
+                    block_state.embeds_list,
+                    block_state.embeds_weights,
+                ) = encode_prompt_items(normalized_prompts)
+
+                # Store prompts_to_encode (which may be target_prompts) for comparison
+                # This ensures we track what was actually encoded
+                block_state.current_prompts = prompts_to_encode
```

Comment:
I feel like the following might be more intuitive given that `current_prompts` is supposed to represent the current prompts in use while `prompts` is supposed to represent the new/next prompts to use.

1. TextConditionBlock sets `prompts` to the prompts to encode (which may be `transition.target_prompts`.
2. EmbeddingBlendingBlock executes the transition (if provided) each time it is called and then when the transition is complete (or if there is no transition) then it would set `current_prompts` to `prompts`.

There would be something like this in EmbeddingBlending block:

```
if not is_transitioning:
  block_state.current_prompts = block_state.prompts
```

And this might allow us to avoid the logic in FrameProcessor for setting prompts to target_prompts?